File: internals/make-hash.rb
content = "what is up, doc?"
header = "blob #{content.length}\0"

store = header + content

require 'digest/sha1'
sha1 = Digest::SHA1.hexdigest(store)
print "SHA hexdigest:\n"
print sha1, "\n"

require 'zlib'
zlib_content = Zlib::Deflate.deflate(store)
print "zlib_content:\n"
print zlib_content.inspect, "\n"

require 'fileutils'
File.open('./zlc_ruby', 'w') { |f| f.write zlib_content }

File: internals/make-hash.py
fin = open("first-file", 'r')
content = fin.read()
print(content)

######content = "what is up, doc?"
header  = "blob {0}\0".format(len(content))
##print(repr(header))

store = header + content
##print(repr(store))

import hashlib
sha1 = hashlib.sha1()

sha1.update(store)
##print("SHA:)
##print(repr(sha1))

sha1_digest = sha1.hexdigest()
print("SHA hexdigest:")
print(repr(sha1_digest))

import zlib
zlib_content = zlib.compress(store)

print("zlib_content:")
print(repr(zlib_content))

fout = open("zlc_python", 'w')
fout.write(zlib_content)
fout.close()


